### ISSUE TITLE
6/27 test.pyについてvideo-classification-3d-cnn-pytorch torch.topk torch.stack

### ISSUE THREAD

**Issue by idekazuki** [idekazuki/diary#27]
```
def calculate_video_results(output_buffer, video_id, test_results, class_names):
    video_outputs = torch.stack(output_buffer)
    average_scores = torch.mean(video_outputs, dim=0)
    sorted_scores, locs = torch.topk(average_scores, k=10)

    video_results = []
    for i in range(sorted_scores.size(0)):
        video_results.append({'label': class_names[locs[i]], 'score': sorted_scores[i]})

    test_results['results'][video_id] = video_results
```
torch.topk について
 torch.topk(input, k, dim=None, largest=True, sorted=True, out=None) -> (Tensor, LongTensor)

    Returns the k largest elements of the given input tensor along a given dimension.

    If dim is not given, the last dimension of the input is chosen.

    If largest is False then the k smallest elements are returned.

    A namedtuple of (values, indices) is returned, where the indices are the indices of the elements in the original input tensor.

    The boolean option sorted if True, will make sure that the returned k elements are themselves sorted

    Parameters

            input (Tensor) – the input tensor

            k (int) – the k in “top-k”

            dim (int, optional) – the dimension to sort along

            largest (bool, optional) – controls whether to return largest or smallest elements

            sorted (bool, optional) – controls whether to return the elements in sorted order

            out (tuple, optional) – the output tuple of (Tensor, LongTensor) that can be optionally given to be used as output buffers

    Example:

    >>> x = torch.arange(1., 6.)
    >>> x
    tensor([ 1.,  2.,  3.,  4.,  5.])
    >>> torch.topk(x, 3)
    torch.return_types.topk(values=tensor([5., 4., 3.]), indices=tensor([4, 3, 2]))

例だと、第一引数のｘに対して大きい順に第２引数で指定した３番目までを抽出している。
largest をFalseにすると小さい順に取り出してくれる。
sortedをFalseにすると結果ををソートせずに出力する。
